model/task_lifecycle.go

```diff
@@ -1055,7 +1056,7 @@ func UpdateBlockedDependencies(ctx context.Context, dependencies []task.Task) er
 		return errors.Wrap(err, "marking unattainable dependencies for tasks")
 	}
 
-	if err := UpdateBlockedDependencies(ctx, dependentTasks); err != nil {
+	if err := UpdateBlockedDependencies(ctx, dependentTasks, true); err != nil {
```

Comment:
Maybe I'm a tad confused, but why would this be true unconditionally rather than passing along `updateAllDependencies`? Maybe it doesn't matter, but it does seem strange enough (since it breaks the recursive pattern) that it warrants a comment explaining why we only obey `updateAllDependencies` for the immediate dependencies and then ignore it for transitive dependencies.

Since this is a highly specific thing to task groups being restarted midway, it would also help to reiterate the reasoning in the function's doc comment about why `updateAllDependencies` only applies to the immediate dependencies and not to transitive dependencies.

Reply:
It definitely should be, thank you.